src/timeline-window.ts

```diff
@@ -193,6 +193,17 @@ export class TimelineWindow {
         return false;
     }
 
+    /**
+     * Notify the window that an event was removed, so we must shrink our
+     * window, if the removed event was inside this window.
+     */
+    public removeEvent(): void {
+        const events = this.getEvents();
+        if (events.length > 0 && events.at(-1) === undefined && this.end) {
```

Comment:
I don't think we can use `.at` - see https://github.com/vector-im/element-web/issues/24245